tutorials/haskell/dotenv-hs/tutorial.md

```diff
@@ -0,0 +1,284 @@
+changing code.
+
+## How it works?
+Dotenv reads the needed application's env vars defined in a *dotenv example file*
+(necesary env vars). If the necesary env vars are defined in *the environment*,
+they stay the same. If they are not there, then dotenv checks if they defined
+in a *dotenv file* and sets all the env vars that are there. Otherwise, it will
+fail, because the necesary env vars are missing. In this way, there is a certanty
+that the configuration of the application is set up before its execution.
+
+# Examples
+The example will be a simple application that prints the content of a table in a
+database. Really simple, but it needs some configuration to run. The standalone
+tool will be used first, and after that, the dotenv library.
+
+[PostgreSQL][https://www.postgresql.org/download/] will be necesary to follow
+this example. Download the next [example code][example] to follow step by step
+the rest of this tutorial.
+
+## Dotenv as an Standalone tool
+First, install the dotenv standalone tool, `stack` or `cabal` can be used:
+
+```shell
+$ stack install dotenv
+```
+
+or
+
+```shell
+$ cabal install dotenv
+```
+
+Once installed, check the help from the tool using the following command
+`dotenv --help`:
+
+```shell
+$ dotenv --help
+
+dotenv - loads options from dotenv files
+
+Usage: dotenv PROGRAM [-e|--dotenv DOTENV] [-x|--dotenv-example DOTENV_EXAMPLE]
+              [-o|--override]
+  Runs PROGRAM after loading options from DOTENV FILE
+
+Available options:
+  -h,--help                Show this help text
+  -e,--dotenv DOTENV       File with the env variables (default: ".env")
+  -x,--dotenv-example DOTENV_EXAMPLE
+                           File with all the necesary env
+                           variables (default: ".env.example")
+  -o,--override            Override existing variables
+```
+
+The minimum necesary is a **PROGRAM** to run. The paths for the *dotenv file*
+and the *dotenv example file* can be set with the short flags `-e` and `-x`,
+respectivelly. The default values are *.env* and *.env.example*, respectivelly.
+The env vars defined in the environment can be overrided with the ones defined
+in the *dotenv file* using the `-o` flag.
+
+Now, go to the directory of the example and follow the next instructions:
+
+Install dependencies and build the project:
+
+```shell
+$ stack build
+```
+
+or
+
+```shell
+$ cabal install --only-dependencies
+$ cabal build
+```
+
+Two executable applications will be created: `dotenv-st` and `dotenv-code`. The
+first one will be used for this part of the example.
+
+Create the databases, one for development and the other one for testing:
+
+```shell
+$ createdb dotenv-dev
+$ createdb dotenv-test
+```
+
+Load the data:
+
+```shell
+$ psql -d dotenv-dev  < sql/dotenv-dev.sql
+$ psql -d dotenv-test < sql/dotenv-test.sql
+```
+
+Before executing the application, check out the *.env.example* file.
+
+```shell
+$ cat .env.example
+
+DB_HOST=
```

Comment:
I think we can simplify the example. The idea is to show the advantages of using our `dotenv` library/standalone tool.